hashi_vault - add support for using mount_point with aws_iam_login auth

briantist opened the pull request:
##### SUMMARY
Fixes #7  

- Allows use of `mount_point` with `aws_iam_auth`

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
hashi_vault.py

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->

briantist commented:
@lamawithonel @ichasco are either of you interested in reviewing this and possibly trying it out in your environment?

lamawithonel commented:
Yeah, I'd be happy to do it, but it might be a week or two.  I'm on vacation untill next week.

briantist commented:
> Yeah, I'd be happy to do it, but it might be a week or two. I'm on vacation until next week.

Thanks @lamawithonel , I have also tested it locally, but it's good to get more feedback if possible, so if you have a chance it's appreciated. We're looking to do a 1.0.0 release of this collection by the end of the month or so, so I'd like to get this change in there if we can!

ichasco commented:
Hi @briantist. How can I test it?

briantist commented:
> Hi @briantist. How can I test it?

This is a good question @ichasco  , and I'm not sure I have the most optimized answer. 
I wonder if @felixfontein has better advice for end users to try out changes in a PR?

Here's something straightforward that isn't the best for all types of changes but should work with this one:
* Create a `lookup_plugins/` directory alongside your playbook
* Copy the version of `hashi_vault.py` in this PR into that directory. Rename it to something distinct, like `test_hashi_vault.py`
* Reference the lookup by the distinct name in the playbook, example:
  ```yaml
  - debug:
      msg: "{{ lookup('test_hashi_vault', 'secret=<etc...>') }}"
  ```

felixfontein commented:
You can also create a folder `collections/ansible_collections/community`, and check out the git repository with this branch in there as the directory `hashi_vault` (i.e. the file galaxy.yml will be in collections/ansible_collections/community/hashi_vault/). Then you can use the lookup `community.hashi_vault.hashi_vault` in your local playbooks. And you can simply delete that directory tree afterwards to stop using this branch.

briantist commented:
@jonnyt this PR is also a change to IAM login, flagging in case it's relevant for you and you have some time to try it out (if not no worries)

lamawithonel reviewed (approved):
I can confirm this works for me.  To test it I replaced the existing `hashi_vault.py` in `site-packages` with this version and deleted the associated `.pyc` file.

briantist commented:
> I can confirm this works for me. To test it I replaced the existing `hashi_vault.py` in `site-packages` with this version and deleted the associated `.pyc` file.

Thank you @lamawithonel !
